### ISSUE TITLE
Automatic trace propagation for Kafka Messages issue

### ISSUE THREAD

**Issue by agolikcorp** [vert-x3/vertx-kafka-client#222]
Hello, 
It is expected that Kafka shared producers add span to traces when a message is being written. It works, but peer.address, peer.hostname and peer.port are always empty.
Producer is created over KafkaProducer.create(vertx, producerConfig) where producerConfig is KafkaClientOptions with configured tracePeerAddress property. 
I assume that it happens because new empty instance of KafkaClientOptions is passed in the line https://github.com/vert-x3/vertx-kafka-client/blob/c5f401087d24458b372d871200072a855bdc02b0/src/main/java/io/vertx/kafka/client/producer/KafkaWriteStream.java#L197

Link to Google Groups Conversation: https://groups.google.com/g/vertx/c/wWbBpOMPmQk/m/KH-_6nuIAAAJ

**Comment by tsegismont** [comment]
@agolikcorp thank you

**Comment by Traderjoe95** [comment]
This issue also triggers unwanted span creation when the tracing policy for a shared producer is set to `IGNORE`. As the default setting is `PROPAGATE`, the Kafka producer will still report a span if there is a currently active span, even though it shouldn't. 

Creating a non-shared producer with a tracing policy is not possible at all, as there is no overloaded version of `KafkaProducer.create` that accepts `KafkaClientOptions`

**Comment by cyhii** [comment]
For now, to workaround this issue, you can create kafka stream and kafka producer(impl) manually, code:

```kotlin
        val kafkaConfig: Map<String, String> = mapOf() // YOUR kafka configs here
        val kafkaClientOptions = KafkaClientOptions.fromMap(kafkaConfig, true)
            .setConfig(kafkaConfig)
            .setTracingPolicy(TracingPolicy.IGNORE)
        val stream: KafkaWriteStream<String, JsonObject> = KafkaWriteStream.create(vertx, kafkaClientOptions)
        return KafkaProducerImpl(vertx, stream).registerCloseHook()
```

**Comment by tsegismont** [comment]
Fixed by #260
